tests/unit/validation/test_validation_functions.py

```diff
@@ -431,6 +434,177 @@ def test_validate_values__infinite_sigmas(sensor_type, parameter):
+            array[parameter[0]] = values[0]
+            array[parameter[1]] = values[1]
+        else:
+            array[parameter[0]][0][0] = values[0][0]
+            array[parameter[0]][0][1] = values[0][1]
+            array[parameter[0]][0][2] = values[0][2]
+            array[parameter[1]][0][0] = values[1][0]
+            array[parameter[1]][0][1] = values[1][1]
+            array[parameter[1]][0][2] = values[1][2]
+        array["id"] = [123, 234, 345]
+
+    sensor_array = initialize_array("input", sensor_type, 3)
+    arbitrary_fill(sensor_array, sensor_type, parameter, values)
+    all_errors = validate_values({sensor_type: sensor_array})
+
+    for error in all_errors:
+        assert any(isinstance(error, error_type) for error_type in error_types)
+        assert set(error.ids).issubset(set(sensor_array["id"]))
+
+
+@pytest.mark.parametrize(
+    ("values", "error_types"),
+    [
+        ([[np.nan, np.nan], [[np.nan, np.nan, np.nan], [np.nan, np.nan, np.nan]]], [InvalidIdError]),
+        (
+            [[0.1, np.nan], [[np.nan, np.nan, np.nan], [np.nan, np.nan, np.nan]]],
+            [InvalidIdError, MultiFieldValidationError],
+        ),
+        (
+            [[np.nan, np.nan], [[np.nan, 0.1, np.nan], [np.nan, np.nan, np.nan]]],
+            [InvalidIdError, MultiFieldValidationError],
+        ),
+        (
+            [[np.nan, np.nan], [[np.nan, 0.1, np.nan], [np.nan, 0.1, np.nan]]],
+            [InvalidIdError, MultiFieldValidationError],
+        ),
+        (
+            [[0.1, 0.1], [[np.nan, np.nan, np.nan], [np.nan, np.nan, np.nan]]],
+            [InvalidIdError, MultiFieldValidationError],
+        ),
+        ([[0.1, 0.1], [[np.nan, np.nan, np.nan], [np.nan, np.nan, np.nan]]], [InvalidIdError]),
+        ([[np.nan, np.nan], [[0.1, 0.1, 0.1], [0.1, 0.1, 0.1]]], [InvalidIdError]),
+        ([[0.1, 0.1], [[0.1, 0.1, 0.1], [0.1, 0.1, 0.1]]], [InvalidIdError]),
+    ],
+)
+def test_validate_values__bad_p_q_sigma_both_components(values, error_types):
+    def two_component_data(values):
+        node = initialize_array("input", "node", 1)
+        node["id"] = 123
+        sym_power_sensor = initialize_array("input", "sym_power_sensor", 1)
+        sym_power_sensor["p_sigma"] = values[0][0]
+        sym_power_sensor["q_sigma"] = values[0][1]
+        sym_power_sensor["id"] = 456
+        asym_power_sensor = initialize_array("input", "asym_power_sensor", 1)
+        asym_power_sensor["p_measured"] = values[1][0]
+        asym_power_sensor["q_measured"] = values[1][1]
+        asym_power_sensor["id"] = 789
+
+        return {
+            "node": node,
+            "sym_power_sensor": sym_power_sensor,
+            "asym_power_sensor": asym_power_sensor,
+        }
+
+    data = two_component_data(values)
+    all_errors = validate_values(data)
+    for error in all_errors:
+        assert any(isinstance(error, error_type) for error_type in error_types)
+        assert data[error.component]["id"][0] == error.ids[0]
```

Comment:
should be like this, right? or am i misunderstanding something?

```suggestion
        assert (data[error.component]["id"] == error.ids).all()
```
or
```suggestion
        for actual, expected in zip(error.ids, data[error.component]["id"]):
            assert actual == expected
```
or
```suggestion
        assert data[error.component]["id"][1] == error.ids[1]
```

Reply:
done